### ISSUE TITLE
WIP on a UI component library

### ISSUE THREAD

**Issue by mbetel** [vugu/vugu#185]
Just wondering how this is progressing, can we get a glimpse?

**Comment by bradleypeabody** [comment]
Interestingly enough, the work that's been done has not yet required anything special yet for things like text fields and form elements - those have been working well as straight HTML and CSS (bootstrap) with simple `:value=`, `@change`, `<option vg-for=...` etc.   Unfortunately I don't have any code I can share right this moment but I will definitely when I can.

Hopefully very soon I'm going to get to work on a data table component, which is where I think is where a more complex Go/Vugu component will really be very useful.

**Comment by Nv7-GitHub** [comment]
I’ve been using Bootstrap with vugu which has allowed me to implement tons of components without much effort - since bootstrap requires basically no JS you can use nearly all bootstrap components in Vugu automatically

**Comment by itstmyi** [comment]
A data table component will be a good example to show complex  and comprehensive code of Vugu . I have tried that to get data / change row color or choosed  checkbox status / edit data / refresh .  I think good code needs enough time and experience, anyway I am really expecting it.

by the way , Vugu is introduced on GitHub Go Wiki (https://github.com/golang/go/wiki/WebAssembly) . It would be nice if it can appear on awesome-go WebAssembly list (https://awesome-go.com/) .

**Comment by bradleypeabody** [comment]
@itstmyi Cool, yes that would be great. Looks like it just needs a PR on the readme here: https://github.com/avelino/awesome-go/blob/master/README.md

**Comment by dcs76** [comment]
I'm trying to push Vugu as the way to go forward for the front end at the company I'm at as I really believe in the idea.

After showing a bit of what it can do to my boss, his initial questions were, 1) is there adoption for it, 2) is there a component library and if there is, can we contribute to it?

I know it is still experimental and that adoption will come once it stabilizes further. I'm really looking forward to the UI library and if you have any information on how to contribute to it would be greatly appreciated.

Thanks for the great work so far.

**Comment by bradleypeabody** [comment]
@dcs76 Thanks and yeah that would be great.

1. I don't have a good way to measure adoption at the moment, but I feel like it has been slowly growing.  I also have a commercial (SaaS) project that has committed to using Vugu for it's UI.  The next big step in this direction is to try to organize a company to officially adopt the maintenance and development, so there is more attention on shoring everything up and moving toward a 1.0 release.  Progress has been slower than I'd hoped, but it is still definitely moving in that direction.

2. There is not really a component library yet, but what's been happening is multiple people have tried using Bootstrap and it's overall worked well and there hasn't been a huge need for a complex wrapper on top of it.  I know for cases like a data table it probably would make sense to have something that wraps such a complex idea and makes it easily usable, but for forms and menus, etc. the most productive thing so far that I've experienced is to just use Bootstrap and raw Vugu.  I also made https://github.com/vugu/vgmbs to help with customizing Bootstrap for a Vugu project.  There is also a more recent example here https://github.com/vugu-examples/taco-store which demonstrates some more complex ideas working together and does not use a component library.  Also, don't get me wrong, I'm not against the idea of having a component library, I just think if one is done it really needs to provide a major clear set of benefits to justify it's existence over just using simple HTML.
